### ISSUE TITLE
Refactor Serial mock away from a singleton

### ISSUE THREAD

**Issue by jacobrosenthal** [ikeyasu/arduino-mock#23]
I have need with several of my libraries to spy on 2 serial at at time, and Arduino has software serial and 2 hardware serial on many boards now. 

Is there a reason Serial mock is a singleton?

**Comment by ikeyasu** [comment]
I just don't know that Serial class could be multiple instances. (I am a user of Uno and Pro Micro.)
You mean we should have Serial1, 2, 3 like http://arduino.cc/en/reference/serial ?

**Comment by jacobrosenthal** [comment]
Sure that exists on arduino so our mock should support it.

Further for me, I have a bunch of Serial tools that Im testing. I have
tools that split the stream into 2 , or combine 2 streams into 1. Tools
that take a stream and print it as hex so its easy to debug.

All of these I need to be able to spy 2 serial at the same time.

**Comment by ikeyasu** [comment]
I may understand that Serial should not be a singleton. If you don't mind, could you give me a sample using multiple Serial instances? Or could you create patches for Serial Mock?

**Comment by jacobrosenthal** [comment]
Hrm. Been playing around for a day or so. I dont necessarily need it not to be a singleton. For Serial1, Serial2 etc can just duplicate the class. I do need it to be a proper Stream class though so it can be passed to things that take Serial... Still trying things.

**Comment by ikeyasu** [comment]
Yeah, I read your implementation for TapStream. I understood that it should not be Singleton.
